presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -150,8 +166,93 @@ public void testColumnsCached()
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 4, 0, 4);
+
+        // Read columns from cache
+        assertThat(cachingJdbcClient.getColumns(firstSession, table)).doesNotContain(phantomColumn);
+        assertThat(cachingJdbcClient.getColumns(secondSession, table)).doesNotContain(phantomColumn);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 4, 2, 4);
+    }
+
+    @Test
+    public void testColumnsCacheInvalidationOnTableDrop()
+    {
+        ConnectorSession firstSession = createSession("first");
+        ConnectorSession secondSession = createSession("second");
+        JdbcTableHandle firstTable = createTable(new SchemaTableName(schema, "first_table"));
+        JdbcTableHandle secondTable = createTable(new SchemaTableName(schema, "second_table"));
+
+        JdbcColumnHandle firstColumn = addColumn(firstTable, "first_column");
+        JdbcColumnHandle secondColumn = addColumn(secondTable, "second_column");
+
+        // Load columns for both tables into cache and assert cache loads (sessions x tables)
+        assertThat(cachingJdbcClient.getColumns(firstSession, firstTable)).contains(firstColumn);
+        assertThat(cachingJdbcClient.getColumns(firstSession, secondTable)).contains(secondColumn);
+        assertThat(cachingJdbcClient.getColumns(secondSession, firstTable)).contains(firstColumn);
+        assertThat(cachingJdbcClient.getColumns(secondSession, secondTable)).contains(secondColumn);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 4, 0, 4);
+
+        // Load columns from cache
+        assertThat(cachingJdbcClient.getColumns(firstSession, firstTable)).contains(firstColumn);
+        assertThat(cachingJdbcClient.getColumns(secondSession, secondTable)).contains(secondColumn);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 4, 2, 4);
+
+        // Drop columns and caches for first table
+        cachingJdbcClient.dropTable(secondSession, firstTable);
+        assertThatThrownBy(() -> cachingJdbcClient.getColumns(firstSession, firstTable)).isInstanceOf(TableNotFoundException.class);
+        assertThatThrownBy(() -> cachingJdbcClient.getColumns(secondSession, firstTable)).isInstanceOf(TableNotFoundException.class);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 6, 2, 6);
+
+        // Check if second table is still cached
+        assertThat(cachingJdbcClient.getColumns(firstSession, secondTable)).contains(secondColumn);
+        assertThat(cachingJdbcClient.getColumns(secondSession, secondTable)).contains(secondColumn);
+        assertCacheLoadsHitsAndMisses(cachingJdbcClient.getColumnsCacheStats(), 6, 4, 6);
+
+        cachingJdbcClient.dropTable(secondSession, secondTable);
+    }
+
+    private void assertCacheLoadsHitsAndMisses(CacheStats stats, int expectedLoad, int expectedHit, int expectedMiss)
```

Comment:
Is there a case where `miss != load`?

Reply:
Not. Would you like me to add that?

Reply:
I was thinking about removing one of these parameters.

Reply:
@wendigo why didn't you want to remove one of those parameters?